contracts/DependencyRegistryV1.sol

```diff
@@ -0,0 +1,1074 @@
+        returns (
+            string memory scriptTypeAndVersion,
+            string memory,
+            uint256 scriptCount
+        ) {
+            return scriptTypeAndVersion;
+        } catch {
+            revert(
+                "Contract does not implement projectScriptDetails and has no override set."
+            );
+        }
+    }
+
+    /**
+     * @notice Converts string to bytes32.
+     * https://ethereum.stackexchange.com/a/9152
+     * @param source string to be converted.
+     * @return result bytes32 representation of string.
+     */
+    function stringToBytes32(
+        string memory source
+    ) public pure returns (bytes32 result) {
+        bytes memory tempEmptyStringTest = bytes(source);
+        if (tempEmptyStringTest.length == 0) {
+            return 0x0;
+        }
+
+        assembly {
+            result := mload(add(source, 32))
+        }
+    }
+
+    /**
+     * @notice Converts bytes to hexadecimal string representation.
+     * https://ethereum.stackexchange.com/a/126928
+     * @param buffer bytes to be converted.
+     * @return string representation of buffer.
+     */
+    function bytesToHexString(
+        bytes memory buffer
+    ) public pure returns (string memory) {
+        // Fixed buffer size for hexadecimal convertion
+        bytes memory converted = new bytes(buffer.length * 2);
+
+        bytes memory _base = "0123456789abcdef";
+
+        for (uint256 i = 0; i < buffer.length; i++) {
+            converted[i * 2] = _base[uint8(buffer[i]) / _base.length];
+            converted[i * 2 + 1] = _base[uint8(buffer[i]) % _base.length];
+        }
+
+        return string(abi.encodePacked("0x", converted));
+    }
+
+    function getTokenHtml(
+        address _contractAddress,
+        uint256 _tokenId
+    ) external view returns (string memory) {
+        uint256 ONE_MILLION = 1_000_000;
+
+        uint256 projectId = _tokenId / ONE_MILLION;
+        bytes32 dependencyType = stringToBytes32(
+            this.getDependencyTypeForProject(_contractAddress, projectId)
+        );
+        bytes32 tokenHash = IDependencyRegistryCompatibleV0(_contractAddress)
+            .tokenIdToHash(_tokenId);
+
+        WrappedScriptRequest[] memory requests = new WrappedScriptRequest[](5);
+
+        requests[0]
+            .scriptContent = 'let css="html{height:100%}body{min-height:100%;margin:0;padding:0}canvas{padding:0;margin:auto;display:block;position:absolute;top:0;bottom:0;left:0;right:0}",head=document.head,style=document.createElement("style");head.appendChild(style),style.type="text/css",style.appendChild(document.createTextNode(css));';
+
+        requests[1].scriptContent = abi.encodePacked(
+            'let tokenData = {"tokenId":"',
+            Strings.toString(_tokenId),
+            '"',
+            ',"hash":"',
+            bytesToHexString(abi.encodePacked(tokenHash)),
+            '"}'
+        );
+
+        requests[2].name = "dependency";
```

Comment:
Note: I initially uploaded just the minified library but ran into gas limit issues so I had to upload a gzipped version instead.

Reply:
curious - do we support uploading in multiple chunks? feels like we should be able to handle arbitrarily large dependencies, even if it means uploading across multiple txs/blocks?

Reply:
To clarify, we do upload dependencies in chunks but the getTokenHtml call was failing to return the full expected output because of provider enforced gas limit (e.g. https://docs.infura.io/infura/networks/ethereum/json-rpc-methods/eth_call)

Reply:
got it, makes sense and I think natural next question would be if we can make it more efficient. Sounds like some recommendations fom @0xdude below will help with that. We could probably use some assembly to convert bytes to string as well, if it enables us to avoid some of these provider-imposed read limits!

Reply:
I'll resolve this thread and address gas efficiency details as separate comment threads!